ui/models/job.js

```diff
@@ -118,35 +140,39 @@ export default class JobModel {
     try {
       notify(`Attempting to retrigger ${jobTerm} via actions.json`, 'info');
 
-      /* eslint-disable no-await-in-loop */
-      for (const id of jobIds) {
-        const job = await JobModel.get(repoName, id);
-        const decisionTaskId = await getGeckoDecisionTaskId(
-          job.push_id,
-          repoName,
-        );
-        const results = await TaskclusterModel.load(decisionTaskId, job);
-        const retriggerTask = results.actions.find(
-          result => result.name === 'retrigger',
-        );
-
-        try {
-          await TaskclusterModel.submit({
-            action: retriggerTask,
-            decisionTaskId,
-            taskId: results.originalTaskId,
-            input: {},
-            staticActionVariables: results.staticActionVariables,
-          });
-        } catch (e) {
-          // The full message is too large to fit in a Treeherder
-          // notification box.
-          notify(formatTaskclusterError(e), 'danger', { sticky: true });
+      const uniquePerPushJobs = {};
+      const jobs = await JobModel.get(repoName, jobIds);
+      for (const job of jobs) {
```

Comment:
This would be better done with ``lodash`` ``groupBy``:

    const uniqueJobsPerPush = groupBy(jobs, job => job.push_id);

Granted, you'd have to get ``job_type_name`` out on line 175, rather than just passing ``value``.